internal/pkg/cli/pipeline_deploy.go

```diff
@@ -179,28 +226,27 @@ func (o *deployPipelineOpts) Execute() error {
 	return nil
 }
 
-func (o *deployPipelineOpts) validatePipelineName() error {
+func (o *deployPipelineOpts) getPipelineMft() (*manifest.PipelineManifest, error) {
 	data, err := o.ws.ReadPipelineManifest()
 	if err != nil {
-		return fmt.Errorf("read pipeline manifest: %w", err)
+		return nil, fmt.Errorf("read pipeline manifest: %w", err)
 	}
 	pipeline, err := manifest.UnmarshalPipeline(data)
 	if err != nil {
-		return fmt.Errorf("unmarshal pipeline manifest: %w", err)
+		return nil, fmt.Errorf("unmarshal pipeline manifest: %w", err)
 	}
-	if pipeline.Name != o.name {
-		return fmt.Errorf(`pipeline %s not found in the workspace`, color.HighlightUserInput(o.name))
+	if err := pipeline.Validate(); err != nil {
+		return nil, fmt.Errorf("validate pipeline: %w", err)
 	}
-	return nil
+	return pipeline, nil
 }
 
 func (o *deployPipelineOpts) convertStages(manifestStages []manifest.PipelineStage) ([]deploy.PipelineStage, error) {
 	var stages []deploy.PipelineStage
-	workloads, err := o.ws.ListWorkloads()
```

Comment:
We probably need better function name for these various workload-listing functions...
Like from what I know there are ones that list ssm-registered workload, ones that that local-workspace workloads (that are not necessarily ssm-registered), and ones that list workloads that are ssm-registered and in the local workspace. It's easy to get these functions mixed up because they have the same or similar names.

Reply:
Also do you think this is rather a `fix` than a `chore`? Would this introduce breaking change?

Reply:
i'll update it to `fix` as this is very obviously a bug fix, we don't consider it a breaking change

Reply:
Added comment to `ListWorkloads` hopefully it would remove ambiguity for a little bit